### PR TITLE
[NEW] Convert Team to Channel

### PR DESCRIPTION
**Pull request by dougfabris**
## Proposed changes (including videos or screenshots)

image

### CODE REVIEW

**Review by KevLehman — APPROVED** [review]
Backend changes looks good. I've also tested overall functionality and it was working as expected.

File: client/views/teams/ConvertToChannelModal/BaseConvertToChannelModal.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	LIST_ROOMS: 'LIST_ROOMS',
+	CONFIRM_CONVERT: 'CONFIRM_CONVERT',
+};
+
+type BaseConvertToChannelModalProps = {
+	onClose: () => void;
+	onCancel: () => void;
+	onConfirm: () => Array<IRoom>;
+	currentStep?: string;
+	rooms: Array<IRoom & { isLastOwner?: string }> | undefined;
```
Comment (tassoevan) [inline client/views/teams/ConvertToChannelModal/BaseConvertToChannelModal.tsx:18]: Why not?

```suggestion
	rooms?: Array<IRoom & { isLastOwner?: string }>;
```

Why is mandatory to pass an `undefined`?

File: client/views/teams/contextualBar/info/TeamsInfoWithData.js [hunk 2]
```diff
@@ -135,8 +140,36 @@ function TeamsInfoWithLogic({ room, openEditing }) {
 
 	const onClickViewChannels = useCallback(() => openTabbar('team-channels'), [openTabbar]);
 
+	const onClickConvertToChannel = useMutableCallback(async () => {
```
[the remark this answers is not in the document]
Comment (dougfabris) [reply client/views/teams/contextualBar/info/TeamsInfoWithData.js:143]: oops!

File: client/views/teams/ConvertToChannelModal/ConvertToChannelModal.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		);
+	}
+
+	return (
+		<BaseConvertToChannelModal
+			onClose={onClose}
+			onCancel={onCancel}
+			onConfirm={onConfirm}
+			rooms={value?.rooms}
+			selectedRooms={{}}
```
Comment (tassoevan) [inline client/views/teams/ConvertToChannelModal/ConvertToChannelModal.tsx:53]: Passing a new object will force re-render every time. Consider using a constant (preferred) or `useMemo()`, if the component is not lightweight enough.

File: client/views/teams/ConvertToChannelModal/ModalSteps/FirstStep.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+			<Box mbe='x24' fontScale='p1'>
+				{t('Notice_that_public_channels_will_be_public_and_visible_to_everyone')}
+			</Box>
+
+			<ChannelDesertionTable
+				lastOwnerWarning={undefined}
+				onToggleAllRooms={onToggleAllRooms}
+				rooms={rooms}
+				params={{}}
```
Comment (tassoevan) [inline client/views/teams/ConvertToChannelModal/ModalSteps/FirstStep.tsx:57]: Passing a new object will force re-render every time. Consider using a constant (preferred) or `useMemo()`, if the component is not lightweight enough.

File: client/views/teams/ConvertToChannelModal/ModalSteps/FirstStep.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+				onChangeParams={(): void => undefined}
```
Comment (tassoevan) [inline client/views/teams/ConvertToChannelModal/ModalSteps/FirstStep.tsx:58]: Passing a new function will force re-render every time. Consider using a constant (preferred) or `useMemo()`, if the component is not lightweight enough.

File: client/views/teams/ConvertToChannelModal/ModalSteps/FirstStep.tsx (hunk 4 above)
Comment (dougfabris) [reply client/views/teams/ConvertToChannelModal/ModalSteps/FirstStep.tsx]: I converted the component to tsx and removed the unnecessary params!

File: client/views/teams/ConvertToChannelModal/ModalSteps/FirstStep.tsx (hunk 5 above)
Comment (dougfabris) [reply client/views/teams/ConvertToChannelModal/ModalSteps/FirstStep.tsx]: I converted the component to tsx and removed the unnecessary params!

**Review by tassoevan — APPROVED** [review]

**Pull request merged**